File: Haskell/Fibonacci/Fibonacci.hs
import System.Environment
import Control.Monad
import Data.Char

main::IO ()
main = do
  args <- getArgs

  when (length args /= 1) (error "Argument is not supplied supplied")

  let nString = head args

  unless (all isDigit nString) (error "Argument should be integer")

  print $ fib $ read nString

fib :: Int -> Int
fib 1 = 1
fib 2 = 1
fib n = fib (n - 1) + fib (n - 2)

File: Java/Fibonacci/src/Fibonacci.java
public class Fibonacci {
	
	public static void main(String[] args) {
		
		long n;
		
		try {
			n = Long.parseLong(args[0]);
		} catch (IndexOutOfBoundsException e) {
			System.out.println("Argument is not supplied");
			return;
		} catch (NumberFormatException e) {
			System.out.println("Argument should be integer");
			return;
		}
		
		long startTime = System.nanoTime();

		System.out.println(fib(n));
		
		System.out.println(System.nanoTime() - startTime);
	}
	
	public static long fib(long n) {
		if(n == 2 || n == 1) {
			return 1;
		} else {
			return fib(n - 1) + fib(n - 2);
		}
	}

}
